content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,169 @@
+
+- **Observation**: At the heart of every effect is its ability to continuously monitor or "observe" signals. 
+This keeps effects primed to detect changes in the signals they are tied to.
+- **Activation**: Upon recognizing a change in an observed signal, the effect will promptly execute its predefined routine. 
+This can involve tasks like updating the UI, triggering a process, or initiating external functions. 
+Through this activation, effects ensure that the system's reactivity is both immediate and relevant.
+
+## Building a reactive system
+
+Reactive systems are designed around the principle of data-driven reactivity. This is when signals and their dependents, effects, maintain a relationship. This relationship ensures a seamless propagation of data changes throughout the system, guaranteeing its responsiveness and real-time updates.
+
+### 1. Initialization
+
+The start of a reactive system begins with the setup of signals. During this phase, signals are initialized with their respective starting values. These initialized signals serve as the primary data sources that the rest of the system will rely on.
+
+### 2. Subscription
+
+Once signals are established, the next step involves the effects indicating their dependencies. 
+In this phase, effects "subscribe" to specific signals, marking their interest and establishing a link. 
+This subscription keeps the effects aware of any changes to the signals they are tied to.
+
+### 3. Notification
+
+During the notification phase, any alteration in a signal's value sets off a chain reaction. 
+The altered signal proactively notifies all its subscribed effects about the modification. 
+Upon receiving this notification, these effects then update or activate as necessary to keep the system's state current and synchronized with the latest data changes.
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity introduces a layer of flexibility and adaptability. 
+Unlike it synchronous counterpart, the updates to the effects are not instantaneous. 
+Instead, they may be delayed or might wait for the completion of an external event or task. 
+This approach is particularly useful in scenarios where immediate updates are not feasible or when the system needs to accommodate external processes before reflecting changes, such as waiting for an API response.
```

Comment:
I asked in the discord:
> Do we have an example of asynchronous reactivity in solid? At first I thought it was referring to batch, but now I think it is referring to suspense. However, batch is more about making sure all state is updated before allowing effects to run, and suspense isn't about delaying computations at all.

In any case, assuming this is referring to batching, I would write it like this:
> This approach is useful in scenarios where an effect depends on multiple signals, and updating one signal before the other would result a data inconsistency. In such a scenario, `batch` can be used to delay triggering effects, so the effect runs after both signals have been updated.

Reply:
do you think its beneficial to talk about both suspense and batch in this case?